update blog contribution policy (#4464)
open-telemetry/opentelemetry.io: content/en/docs/contributing/blog.md

The [OpenTelemetry blog](/blog/) communicates new features, community reports,
and any news that might be relevant to the OpenTelemetry community. This

> includes end users and developers. Anyone can write a blog post, read below what
> the requirements are.

Reviewer: ```suggestion
includes end users and developers. Anyone can write a blog post, read the following
requirements to get started.
```

- Vendor product pitches

If your blog post fits into the list of appropriate content,
[raise an issue](https://github.com/open-telemetry/opentelemetry.io/issues/new?title=New%20Blog%20Post:%20%3Ctitle%3E)
with the following details:

- Title of the blog post
- Short description and outline of your blog post
- If applicable, list the technologies used in your blog post. Make sure that

>   all of them are open source, and prefer CNCF projects over non-CNCF projects
>   (e.g. use Jaeger for trace visualization, and Prometheus for metric
>   visualization)

Reviewer: ```suggestion
  all of them are open source, and prefer CNCF projects over non-CNCF projects. For example, use Jaeger for trace visualization, and Prometheus for metric
  visualization)
```

- Name of a [SIG](https://github.com/open-telemetry/community/), which is
  related to this blog post

> - Name of a sponsor (maintainer or approver) from this SIG, who will help to

Reviewer: If there's a neutrality concern, be mindful that  contributors might still nominate a SIG member from the same company. Of course we expect SIG members to be good citizens. :)

Author: Good call out, we can add an "ideally from a different company"

> - Name of a sponsor (maintainer or approver) from this SIG, who will help to

Reviewer 2: The previous step requires that a SIG be identified, which is a great step forward. I'd suggest that they cc the SIG approvers in the issue. We docs-approvers can then ensure that the corresponding SIG approvers are listed as require approvers for the PR.

So, I might be tempted to suggest that **finding a sponsor be _optional_** (for now), but state that having a sponsor _may_ increase their chances of having their blog reviewed and approved more quickly (or some such language). WDYT?

Author: I agree, keeping it optional for now makes our life easier as well for a start, since we do not have to push back that hard (based on our policy) but can let people know that we see their PR but it'S not our prio right now and they should get a sponsor to help themselves
